cmd/depth/depth.go

```diff
@@ -20,45 +21,44 @@ const (
 var outputJSON bool
 
 func main() {
-	t := parse(os.Args[1:])
-	if err := handlePkgs(t, flag.Args(), outputJSON); err != nil {
-		os.Exit(1)
+	ResolveInternal := flag.Bool("internal", false, "If set, resolves dependencies of internal (stdlib) packages.")
+	ResolveTest := flag.Bool("test", false, "If set, resolves dependencies used for testing.")
+	MaxDepth := flag.Int("max", 0, "Sets the maximum depth of dependencies to resolve.")
+	outputJSON := flag.Bool("json", false, "If set, outputs the depencies in JSON format.")
+	flag.Parse()
+
+	pkgs := flag.Args()
+	t := &depth.Tree{
+		ResolveInternal: *ResolveInternal,
+		ResolveTest:     *ResolveTest,
+		MaxDepth:        *MaxDepth,
 	}
-}
-
-// parse constructs a depth.Tree from command-line arguments.
-func parse(args []string) *depth.Tree {
```

Comment:
Now that you're using the WaitGroup and not channels I think you can keep the parse method + test